From 2e67a4f1b4d9539054a3ed797d841627db4e1f58 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Wed, 30 Jul 2025 12:42:52 +0200
Subject: [PATCH] drm/bridge: synopsys: dw-mipi-dsi: remove property default-on

To obtain a continuous display after the probe, do not reset the DSI bridge
if it is powered on (default-on property not define by bindings).

Change-Id: I6f4859177abaaab477ff2cfccfb51cb41dd0a2b0
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/479226
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/gpu/drm/bridge/synopsys/dw-mipi-dsi.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

--- a/drivers/gpu/drm/bridge/synopsys/dw-mipi-dsi.c
+++ b/drivers/gpu/drm/bridge/synopsys/dw-mipi-dsi.c
@@ -1175,6 +1175,7 @@ __dw_mipi_dsi_probe(struct platform_devi
 	struct reset_control *apb_rst;
 	struct dw_mipi_dsi *dsi;
 	int ret;
+	u32 pwr;
 
 	dsi = devm_kzalloc(dev, sizeof(*dsi), GFP_KERNEL);
 	if (!dsi)
@@ -1209,7 +1210,12 @@ __dw_mipi_dsi_probe(struct platform_devi
 	 * Note that the reset was not defined in the initial device tree, so
 	 * we have to be prepared for it not being found.
 	 */
-	if (!device_property_read_bool(dev, "default-on")) {
+	pwr = dsi_read(dsi, DSI_PWR_UP);
+	/*
+	 * To obtain a continuous display after the probe,
+	 * do not reset the DSI bridge if it is powered on.
+	 */
+	if (pwr != POWERUP) {
 		apb_rst = devm_reset_control_get_optional_exclusive(dev, "apb");
 		if (IS_ERR(apb_rst)) {
 			ret = PTR_ERR(apb_rst);
